### PR TITLE
bpo-33228: Use Random.choices in tempfile

### PR DESCRIPTION
**Pull request by wm75**
Using Random.choices instead of Random.choice in
the tempfile._RandomNameSequence class makes the code more
readable and faster.
This commit also updates the docstring of the class.

https://bugs.python.org/issue33228

### CODE REVIEW

File: Lib/tempfile.py [hunk 1]
```diff
@@ -151,9 +151,7 @@ def __iter__(self):
         return self
 
     def __next__(self):
-        c = self.characters
-        choose = self.rng.choice
-        letters = [choose(c) for dummy in range(8)]
+        letters = self.rng.choices(self.characters, k=8)
```
Comment (rhettinger) [inline Lib/tempfile.py:154]: I don't think these kinds of trivial changes should be done.  There should be a preference for code stability.

**Comment by 1st1** [comment]
How much "faster" does it make the code? Do you have a use case for which you need to optimize tempfile?

**Comment by wm75** [comment]
Well, _RandomNameSequence would get a bit more than 2x faster, but, of course, in the context of tempfile almost all public functions will be IO limited anyway, and the performance gain in this one class will be completely irrelevant.
No, I don't have a use case that needs this optimization. Rather I was coming across this, while working on #6291, when I was checking where in the stdlib random.Random gets used.
If you think this is not worth the effort, feel free to just close it, I won't mind :smiley:

**Comment by wm75** [comment]
@methane, I split out the docstring change into a separate PR #6414 like you asked for.

@rhettinger @1st1 if the remainder of this PR is not worth it, it can be closed without any loss now.

**Pull request closed without merging**